python_modules/dagster/dagster/_core/definitions/asset_graph.py

```diff
@@ -0,0 +1,30 @@
+from typing import Sequence, Union
+
+import dagster._check as check
+from dagster._core.selector.subset_selector import (
+    generate_asset_dep_graph,
+    generate_asset_name_to_definition_map,
+)
+
+from .assets import AssetsDefinition
+from .source_asset import SourceAsset
+
+
+class AssetGraph:
+    def __init__(self, all_assets: Sequence[Union[AssetsDefinition, SourceAsset]]):
+        assets_defs = []
+        source_assets = []
+        for asset in all_assets:
+            if isinstance(asset, SourceAsset):
+                source_assets.append(asset)
+            elif isinstance(asset, AssetsDefinition):
+                assets_defs.append(asset)
+            else:
+                check.failed(f"Expected SourceAsset or AssetsDefinition, got {type(asset)}")
+
+        self.assets_defs = assets_defs
+        self.asset_dep_graph = generate_asset_dep_graph(assets_defs, source_assets)
```

Comment:
to be honest, I don't remember why we keep this dependency graph in the language of asset key strings after we create it. is this necessary?